lazyscribe/artifacts/__init__.py

```diff
@@ -23,11 +24,30 @@ def _get_handler(alias: str) -> Type[Artifact]:
         The artifact handler class object. This object will need to be constructed
         using :py:meth:`lazyscribe.artifacts.Artifact.construct`.
     """
-    for obj in Artifact.__subclasses__():
-        if obj.alias == alias:
-            out = obj
+    eps = entry_points()
+    entry = eps.select(group="artifact_type")
+
+    for full_artifact_class in entry:
+        if full_artifact_class.name == alias:
+            mod, name = full_artifact_class.value.rsplit(".", 1)
+            try:
+                mod = import_module(mod)
+            except ImportError as imp:
+                raise RuntimeError(
+                    f"Unable to import handler for {alias} through entry points or standard import."
+                ) from imp
+
+            for part in name.split("."):
+                mod = getattr(mod, part)
```

Comment:
Using `rsplit` always feels a bit error prone (to me). I was playing around with this, what do you think? We can modify the `pyproject.toml` to specify `<path>.<to>:<HandlerClass>`:

```toml
[project.entry-points."artifact_type"]
base = "lazyscribe.artifacts.base:Artifact"
joblib = "lazyscribe.artifacts.joblib:JoblibArtifact"
json = "lazyscribe.artifacts.json:JSONArtifact"
testartifact = "tests.conftest:TestArtifact"
```

then we can use the `load()` method from entry points:

```suggestion
        try:
            mod = full_artifact_class.load()
        except ImportError as imp:
            raise RuntimeError(
                f"Unable to import handler for {alias} through entry points."
            ) from imp
```

I just did a quick look at it in Python 3.10, so not sure if it's a solution we can roll with fully.